server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -868,6 +1016,14 @@ public Query termsQuery(String field, List<Object> values) {
                     v = Arrays.copyOf(v, upTo);
                 }
 
+                if (isSearchable && hasDocvalues) {
+                    Query query = BigIntegerPoint.newSetQuery(field, v);
+                    Query dvQuery = SortedNumericDocValuesField.newSlowSetQuery(field, points);
```

Comment:
Usage of `SortedNumericDocValuesField.newSlowSetQuery(field, points);` does not seem right here: we are dealing with unsigned long here, where as  `points` represent a regular `long` value (signed long), the query will be incorrect

Reply:
Thanks for bringing this up @reta, I was wondering if lucene supports a `slowSetQuery` for BigInteger?

Reply:
> Thanks for bringing this up @reta, I was wondering if lucene supports a `slowSetQuery` for BigInteger?

@harshavamsi I don't think so, had this dilemma with range queries, ended up with `SortedUnsignedLongDocValuesRangeQuery`: the comparison functions used there rely on `Long.compareUnsigned`

Reply:
@reta thanks for the suggestion. I replicated this with a custom implementation of `SortedUnsignedLongDocValuesSetQuery`. It takes in an array of `BigInts`, get the long values and uses `Long.compareUnsigned` to do comparisons. I had to introduce the `LongHashSet` lucene class since it was package private. I am now running into a JarHell exception since this class exists in both places. I couldn't find a way to resolve this. Do you know anything about this?